example.py

```diff
@@ -9,15 +9,21 @@ class ExampleLoggingApp(krux.cli.Application):
       * a command-line option parser
       * a logger
     """
-    def __init__(self):
+    def __init__(self, *args, **kwargs):
         # the 'name' parameter is just a string/name for your script. It should be unique in your environment,
         # because stats will be emitted to statsd with that as part of the stat name.
 
         # the syslog_facility parameter being set changes where logs get sent to;
         # the default is to print them out to the console; setting a syslog facility
         # means messages will go to syslog with the named facility and called severity;
         # your local syslog config will need to route those messages to a file or remote syslog.
-        super(ExampleLoggingApp, self).__init__(name='testloggingapp', syslog_facility='local0')
+
+        # A NOTE ON *args AND **kwargs: Our intent is to capture and pass on the arguments
+        # to parent methods, which want to use them. This is one way. Another way
+        # is to declare the arguments explicitly. There's a software maintenance cost to that, though:
+        # if parameter changes are made to a ancestor method, it may require a change
+        # to the descendant methods.
+        super(ExampleLoggingApp, self).__init__(name='testloggingapp', syslog_facility='local0', *args, **kwargs)
```

Comment:
I strongly disagree with encouraging the use of `*args, **kwargs` in end-user code. Your note is good, but overlooks the maintenance cost that comes with using `*args, **kwargs`, when you specify your arguments explicitly, your exceptions tell you exactly what went wrong:

```python
def explicit_arguments(arg1, arg2):
    print('arg1: {}'.format(arg1))
    print('arg2: {}'.format(arg2))

if __name__ == '__main__':
    explicit_arguments('foo')
```
```
14:22 $ ./example.py
Traceback (most recent call last):
  File "./example.py", line 15, in <module>
    explicit_arguments('foo')
TypeError: explicit_arguments() takes exactly 2 arguments (1 given)
```

"Oops, I passed the wrong number of arguments to that library function!". Alternately:

```python
def implicit_arguments(*args, **kwargs):
    print('arg1: {}'.format(args[0]))
    print('arg2: {}'.format(args[1]))

if __name__ == '__main__':
    implicit_arguments('bar')
```
```
14:22 $ ./example.py
arg1: bar
Traceback (most recent call last):
  File "./example.py", line 15, in <module>
    implicit_arguments('bar')
  File "./example.py", line 11, in implicit_arguments
    print('arg2: {}'.format(args[1]))
IndexError: tuple index out of range
```

"Darn library author totally screwed up! Their code is buggy! python-krux-stdlib is for hosers! PEC team, fix your code!"

Explicit > Implicit